src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs

```diff
@@ -169,7 +168,9 @@ public bool TryGetTelemetryId(out Guid telemetryId)
                     //       IWaitIndicator abstraction (which is a thin wrapper on top of IVsThreadedWaitDialog) directly
                     //       for now, we use the one LB created before calling us. meaning we don't update
                     //       text on the wait dialog window
-                    _workspaceStatusService.WaitUntilFullyLoadedAsync(cancellationToken).Wait(cancellationToken);
+                    // 
+                    //       this also needs to run under threading context otherwise, we can deadlock on VS
+                    ThreadingContext.JoinableTaskFactory.Run(() => _workspaceStatusService.WaitUntilFullyLoadedAsync(cancellationToken));
                 }
```

Comment:
by the way, why we leak that we are using JTF in threadingcontext? it could just expose ThreadingContext.Wait so we just use it for every case we use Wait or Result?